api/graphql/types/recurring_reservation/serializers.py

```diff
@@ -126,3 +134,241 @@ class Meta(RecurringReservationCreateSerializer.Meta):
+        with transaction.atomic():
+            instance = super().save()
+            self.create_reservations(instance, reservation_type, reservation_user, skip, check_opening_hours)
+
+        return instance
+
+    def create_reservations(
+        self,
+        instance: RecurringReservation,
+        reservation_type: ReservationTypeChoice,
+        reservation_user: User,
+        skip: list[datetime.date],
+        check_opening_hours: bool,
+    ) -> None:
+        reserved_timespans = self.get_reserved_timespans(
+            start_date=instance.begin_date,
+            end_date=instance.end_date,
+            reservation_unit=instance.reservation_unit,
+        )
+
+        reservable_timespans = self.get_reservable_timespans(instance) if check_opening_hours else []
+
+        non_overlapping: list[ReservationPeriod] = []
+        overlapping: list[ReservationPeriodJSON] = []
+        not_reservable: list[ReservationPeriodJSON] = []
+
+        begin_time: datetime.time = instance.begin_time  # type: ignore[assignment]
+        end_time: datetime.time = instance.end_time  # type: ignore[assignment]
+
+        weekdays: list[int] = [int(val) for val in instance.weekdays.split(",")]
+        if not weekdays:
+            weekdays = [instance.begin_date.weekday()]
+
+        for weekday in weekdays:
+            delta: int = weekday - instance.begin_date.weekday()
+            if delta < 0:
+                delta += 7
+
+            begin_date: datetime.date = instance.begin_date + datetime.timedelta(days=delta)
+
+            periods = get_periods_between(
+                start_date=begin_date,
+                end_date=instance.end_date,
+                start_time=begin_time,
+                end_time=end_time,
+                interval=instance.recurrence_in_days,
+                tzinfo=DEFAULT_TIMEZONE,
+            )
+            for begin, end in periods:
+                if begin.date() in skip:
+                    continue
+
+                timespan = TimeSpanElement(
+                    start_datetime=begin,
+                    end_datetime=end,
+                    is_reservable=True,
+                    buffer_time_after=instance.reservation_unit.buffer_time_after,
+                    buffer_time_before=instance.reservation_unit.buffer_time_before,
+                )
+
+                # Would the reservation overlap with any existing reservations?
+                # Also checks that buffers for the reservation will also not overlap.
+                if any(timespan.overlaps_with(reserved) for reserved in reserved_timespans):
+                    overlapping.append(
+                        ReservationPeriodJSON(
+                            begin=begin.isoformat(timespec="seconds"),
+                            end=end.isoformat(timespec="seconds"),
+                        )
+                    )
+                    continue
```

Comment:
> Also checks that buffers for the reservation will also not overlap

This does not check the buffers for the existing reservations.
The `overlaps_with` method does check the buffers of the calling timespan (`timespan`), uses the buffers of the other timespan (`reserved`)
Above you set the `buffer_time_after and `..._before` to `timezone`, but the timespans returned from `get_reserved_timespans` in `reserved_timespans` don't have any buffer times set.

You should also check if any of the `reserved` overlap with with `timespan` to make sure the buffers don't overlap that way either.

Reply:
After using `get_affecting_reservations_as_closed_time_spans`, I'm checking overlap like this:

```python
# Would the reservation timespan overlap with any closing timespans
# that exist due to existing reservations? Checks for:
# 1) Unbuffered reservation timespan overlapping with any buffered closed timespan
# 2) Unbuffered closed timespan overlapping with any buffered reservation timespan
# 3) Unbuffered reservation timespan overlapping with any unbuffered blocked timespan
if (
    any(reservation_timespan.overlaps_with(closed) for closed in closed_timespans)
    or any(closed.overlaps_with(reservation_timespan) for closed in closed_timespans)
    or any(reservation_timespan.overlaps_with(blocked) for blocked in blocked_timespans)
):
    results.overlapping.append(ReservationPeriod(begin=begin, end=end))
    continue
```